apps/web/src/components/workflow/FlowEditor.tsx

```diff
@@ -332,9 +327,17 @@ export function FlowEditor({
              * Remove this line once we tackle the workflow state handling
              */
             deleteKeyCode={null}
-            {...reactFlowDefaultProps}
+            defaultEdgeOptions={defaultEdgeOptions}
+            zoomOnScroll={zoomOnScroll}
+            preventScrolling={preventScrolling}
+            nodesConnectable={nodesConnectable}
+            nodesDraggable={nodesDraggable}
+            minZoom={minZoom}
+            maxZoom={maxZoom}
+            defaultZoom={defaultZoom}
+            {...restProps}
           >
-            <Controls />
+            {withControls && <Controls />}
```

Comment:
Thinking if we should pass `withControls` as a `show` property to the Controls component and the proper component to handle their rendering rather than the parent. What do you think?

Reply:
This `Controls` component is from the `react-flow-renderer` library and it doesn't have the `show` prop.
But still, I think it's better to do it on the parent level, otherwise, if you do it on the child level then you might register some hooks unnecessarily and the React will create a fiber for that component and will do all its logic aka reconciliation work.